logbook/handlers.py

```diff
@@ -867,29 +868,61 @@ class TimedRotatingFileHandler(FileHandler):
 
     By default it will keep all these files around, if you want to limit
     them, you can specify a `backup_count`.
+
+    You may supply an optional `rollover_format`. This allows you to specify
+    the format for the filenames of rolled-over files.
+    the format as
+
+    So for example if you configure your handler like this::
+
+        handler = TimedRotatingFileHandler(
+            '/var/log/foo.log',
+            date_format='%Y-%m-%d',
+            rollover_format='{basename}{ext}.{timestamp}')
+
+    The filenames for the logfiles will look like this::
+
+        /var/log/foo.log.2010-01-10
+        /var/log/foo.log.2010-01-11
+        ...
+
+    Finally, an optional argument `timed_filename_for_current` may be set to
+    false if you wish to have the current log file match the supplied filename
+    until it is rolled over
     """
 
     def __init__(self, filename, mode='a', encoding='utf-8', level=NOTSET,
                  format_string=None, date_format='%Y-%m-%d',
-                 backup_count=0, filter=None, bubble=False):
-        FileHandler.__init__(self, filename, mode, encoding, level,
-                             format_string, True, filter, bubble)
+                 backup_count=0, filter=None, bubble=False,
+                 timed_filename_for_current=True,
+                 rollover_format='{basename}-{timestamp}{ext}'):
         self.date_format = date_format
         self.backup_count = backup_count
-        self._fn_parts = os.path.splitext(os.path.abspath(filename))
-        self._filename = None
-
-    def _get_timed_filename(self, datetime):
-        return (datetime.strftime('-' + self.date_format)
-                .join(self._fn_parts))
-
-    def should_rollover(self, record):
-        fn = self._get_timed_filename(record.time)
-        rv = self._filename is not None and self._filename != fn
-        # remember the current filename.  In case rv is True, the rollover
-        # performing function will already have the new filename
-        self._filename = fn
-        return rv
+
+        self.rollover_format = rollover_format
+
+        self.original_filename = filename
+        self.basename, self.ext = os.path.splitext(os.path.abspath(filename))
+        self.timed_filename_for_current = timed_filename_for_current
+
+        self._timestamp = self.get_timestamp(_datetime_factory())
+        timed_filename = self.compute_name(self._timestamp)
+
+        if self.timed_filename_for_current:
+            filename = timed_filename
+
+        FileHandler.__init__(self, filename, mode, encoding, level,
+                             format_string, True, filter, bubble)
+
+    def get_timestamp(self, datetime):
+        return datetime.strftime(self.date_format)
+
+    def compute_name(self, timestamp):
```

Comment:
I would call it `format_name` or `generate_name`. Compute is generally used to refer to numeric calculations...